lib/backend/k8s/conversion.go

```diff
@@ -354,23 +370,29 @@ func (c Converter) k8sIngressRuleToCalico(r extensions.NetworkPolicyIngressRule,
 func (c Converter) buildRule(port *extensions.NetworkPolicyPort, peer *extensions.NetworkPolicyPeer, ns string) model.Rule {
 	var protocol *numorstring.Protocol
 	var dstPorts []numorstring.Port
-	srcSelector := ""
+	var (
+		selector string
+		nets     []*cnet.IPNet
+		notNets  []*cnet.IPNet
+	)
 	if port != nil {
 		// Port information available.
 		protocol = c.k8sProtocolToCalico(port.Protocol)
 		dstPorts = c.k8sPortToCalico(*port)
 	}
 	if peer != nil {
 		// Peer information available.
-		srcSelector = c.k8sPeerToCalicoSelector(*peer, ns)
+		selector, nets, notNets = c.k8sPeerToCalicoSrcFields(*peer, ns)
 	}
 
 	// Build the rule.
 	return model.Rule{
 		Action:      "allow",
 		Protocol:    protocol,
-		SrcSelector: srcSelector,
+		SrcSelector: selector,
```

Comment:
Hm, I don't think this works actually, since this function gets called for both ingress and egress rule generation.

Don't we need to set `SrcSelector` / `DstSelector` accordingly?

Same for the `Nets` fields.

Reply:
Indeed, good catch.

Reply:
But interestingly, ports are always DstPorts, in both ingress and egress rules.